fastddsspy_tool/test/application/test_class.py

```diff
@@ -0,0 +1,279 @@
+        id_rate = rate[rate.find('rate:'):]
+        if re.match(pattern_1, id_rate):
+            return True
+        if re.match(pattern_2, id_rate):
+            return True
+        print('Not valid rate: ')
+        print(rate)
+        return False
+
+    def valid_output(self, output) -> bool:
+        """
+        @brief Check the validity of the output against the expected output.
+
+        @param output: The actual output obtained from executing a command.
+        @return Returns True if the output matches the expected output or \
+        satisfies specific conditions for lines containing '%%guid%%' or '%%rate%%', \
+        False otherwise.
+
+        The function compares the provided 'output' with the expected output.
+        It checks each line of the 'output' and 'expected_output' to determine their validity.
+
+        If the entire 'output' matches the 'expected_output', the function returns True.
+        Otherwise, it iterates over each line and performs the following checks:
+        - If a line in 'expected_output' contains '%%guid%%', it calls the 'valid_guid()'.
+        - If a line in 'expected_output' contains '%%rate%%', it calls the 'valid_rate()'.
+        - If a line does not contain '%%guid%%' or '%%rate%%', it compares the corresponding \
+            lines in 'output' and 'expected_output' for equality.
+
+        If any line does not meet the expected conditions, the function returns False and prints \
+        the 'output' and 'expected_output' for debugging purposes.
+
+        """
+        expected_output = self.output_command()
+        if expected_output == output:
+            return True
+
+        lines_expected_output = expected_output.splitlines()
+        lines_output = output.splitlines()
+        guid = True
+        rate = True
+        for i in range(len(lines_expected_output)):
+            if '%%guid%%' in lines_expected_output[i]:
+                guid = self.valid_guid(lines_output[i])
+            elif '%%rate%%' in lines_expected_output[i]:
+                rate = self.valid_rate(lines_output[i])
+            elif lines_expected_output[i] != lines_output[i]:
+                print('Output: ')
+                print(output)
+                print('Expected output: ')
+                print(expected_output)
+                return False
+        return (guid and rate)
+
+    def stop_tool(self, proc) -> int:
+        """
+        @brief Stop the running Spy.
+
+        @param proc: The subprocess object representing the running Spy.
+        @return Returns 1 if the Spy is successfully stopped, 0 otherwise.
+        """
+        try:
+            proc.communicate(input='exit\n', timeout=10)[0]
+        except subprocess.TimeoutExpired:
+            proc.kill()
+
+        time.sleep(SLEEP_TIME)
+
+        if not self.is_stop(proc):
+            print('ERROR: DDS Spy still running')
+            return 0
+
+        return 1
+
+    def stop_dds(self, proc) -> int:
```

Comment:
NIT: Be careful with using `int` as a return value.
If you want to differentiate between ok and error, use a boolean. If you want more complex return codes, use enumerations.
The main problem here is that OK for a process stop is 0, not 1. So this may be confusing.
(I give you the point that is well documented though, but try to avoid this).